types/office-js-preview/index.d.ts

```diff
@@ -11213,6 +11213,61 @@ declare namespace Office {
      * **{@link https://docs.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose or Read
      */
     interface Body {
+        /**
+         * On send, adds the specified content to the end of the item body, after any signature.
+         * 
+         * **Important**: To use `appendOnSendAsync`, the `AppendOnSend` extended permission must be included in the `ExtendedPermissions` node of the manifest.
+         * 
+         * [Api set: Mailbox Preview]
+         * 
+         * @remarks
+         * 
+         * **{@link https://docs.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: `ReadWriteItem`
+         * 
+         * **{@link https://docs.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose
+         * 
+         * **Errors**:
+         * 
+         * - `DataExceedsMaximumSize`: The data parameter is longer than 1,000,000 characters.
+         * 
+         * - `InvalidFormatError`: The `options.coercionType` parameter is set to `Office.CoercionType.Html` but the `data` parameter is in plain text.
+         * 
+         * @param data - The string to be added to the end of the body. The string is limited to 1,000,000 characters.
+         * @param options - Optional. An object literal that contains one or more of the following properties.
+         *        `asyncContext`: Developers can provide any object they wish to access in the callback method.
+         *        `coercionType`: The desired format for the data to be appended. The string in the `data` parameter will be converted to this format.
```

Comment:
The coercionType I don't think is optional. @timwan to comment more

Reply:
unless it's a separate parameter, if it's part of options, seems like it should be optional

Reply:
or are you saying that if options is included, coercionType is required?

Reply:
Ankush confirmed that options.coercionType is optional